libs/ts4wing/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import { std, platform } from "@winglang/sdk";
+import { Construct } from "constructs";
+
+/**
+ * Properties for a Wing app.
+ */
+export interface WingAppProps {
+  /**
+   * The name and id of the app.
+   * @default "main"
+   */
+  name?: string;
+}
+
+/**
+ * Create a Wing app.
+ *
+ * ```ts
+ * import { wing } from "ts4wing";
+ * import { cloud } from "@winglang/sdk";
+ *
+ * wing(app => {
+ *   new cloud.Bucket(app, "Bucket");
+ * });
+ * ```
+ *
+ * @param fn Define your application using the provided root construct.
+ *           Note that this function may be called multiple times when used with `wing test`.
+ */
+export function wing(fn: (root: Construct) => void, props: WingAppProps = {}) {
+  // check if we have everything we need
+  const requiredEnvVars = [
+    "WING_PLATFORMS",
+    "WING_SYNTH_DIR",
+    "WING_SOURCE_DIR",
+    "WING_IS_TEST",
+  ];
+  for (const envVar of requiredEnvVars) {
+    if (process.env[envVar] === undefined) {
+      throw new Error(`\
+Missing environment variable: ${envVar}
+This is a Wing app and must be run through the Wing CLI (npm install -f winglang).`);
+    }
+  }
+
+  class $Root extends std.Resource {
+    constructor(scope: Construct, id: string) {
+      super(scope, id);
+      fn(this);
+    }
+  }
+
+  const platformPaths = ((s) => (!s ? [] : s.split(";")))(
+    process.env.WING_PLATFORMS
+  );
+  const outdir = process.env.WING_SYNTH_DIR;
```

Comment:
Seems like this is a duplicated code from the generated wing code. Maybe we can reuse in the future